src/app/forms/components/defect/defect.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, Input } from '@angular/core';
+import { FormNode } from '@forms/services/dynamic-form.types';
+import { DefectTpl } from '@forms/templates/general/defect.template';
+import { Defect } from '@models/defect';
+import { DefectAdditionalInformationLocation } from '@models/defectAdditionalInformationLocation';
+
+@Component({
+  selector: 'app-defect',
+  templateUrl: './defect.component.html',
+  styleUrls: ['./defect.component.scss']
+})
+export class DefectComponent {
+  @Input() edit = false;
+  @Input() defectData!: Defect;
+
+  defectTpl: FormNode;
+
+  constructor() {
+    this.defectTpl = DefectTpl;
+  }
+
+  mapLocationText(location: DefectAdditionalInformationLocation) {
```

Comment:
Can we have a comment here, it's not obvious what this is doing

Reply:
Done.